utilities/src/main/java/io/syndesis/qe/bdd/validation/CommonValidationSteps.java

```diff
@@ -40,4 +44,68 @@ public void waitForIntegrationToBeActive(String integrationName) {
+		log.info("There is build with name {} running", sanitizedName);
+	}
+
+	@Then(value = "^verify there are no s2i builds running for integration: \"([^\"]*)\"$")
+	public void verifyNoIntegrationBuildRunning(String integrationName) {
+		final String sanitizedName = integrationName.toLowerCase().replaceAll(" ", "-");
+		Assertions.assertThat(new ArrayList<>(OpenShiftUtils.getInstance().getBuilds())).filteredOn(build -> build.getMetadata().getLabels().get("buildconfig").contentEquals(sanitizedName)).isEmpty();
+		log.info("There is no builds with name {} running", sanitizedName);
+	}
+
+	@Then("^verify integration \"([^\"]*)\" has current state \"([^\"]*)\"")
+	public void verifyIntegrationState(String integrationName, String integrationState) {
+		final Integration integration = integrationsEndpoint.list().stream()
+				.filter(item -> item.getName().equals(integrationName))
+				.collect(Collectors.toList()).get(0);
+
+		log.debug("Actual state: {} and desired state: {}", integration.getCurrentStatus().get().name(), integrationState);
+		Assertions.assertThat(integration.getCurrentStatus().get().name()).isEqualTo(integrationState);
+	}
+
+	@Then("^validate integration: \"([^\"]*)\" pod scaled to (\\d+)$")
+	public void verifyPodCount(String integrationName, int podCount) {
+		final String sanitizedName = integrationName.toLowerCase().replaceAll(" ", "-");
+		try {
+			final String errorMessage = "Wrong number of pods " + sanitizedName;
+			OpenShiftWaitUtils.assertEventually(errorMessage, OpenShiftWaitUtils.areExactlyNPodsRunning("deploymentconfig", sanitizedName, podCount));
+		} catch (InterruptedException ex) {
+			log.error("Error: {}", ex);
+		}
+		log.info("There are {} pods with name {} running", podCount, sanitizedName);
+	}
+
+	@Then("^switch Inactive and Active state on integration \"([^\"]*)\" for (\\d+) times and check pods up/down")
+	public void verifyIntegrationOnOffNTimes(String integrationName, int switchNTimes) {
+
+		for (int i = 0; i <= switchNTimes; i++) {
+
+			final IntegrationDeploymentState newDepState;
+			final Integration integration = integrationsEndpoint.list().stream().filter(a -> a.getName().contentEquals(integrationName)).findFirst().get();
+			if (integration.getCurrentStatus().get().equals(IntegrationDeploymentState.Active)) {
+				newDepState = IntegrationDeploymentState.Inactive;
+			} else {
+				newDepState = IntegrationDeploymentState.Active;
+			}
+
+			final Integration updatedIntegration = new Integration.Builder().createFrom(integration)
+					.desiredStatus(newDepState).build();
+			integrationsEndpoint.update(integration.getId().get(), updatedIntegration);
+
+			if (newDepState.name().equals("Active")) {
+				verifyPodCount(integrationName, 1);
+			} else {
+				verifyPodCount(integrationName, 0);
+			}
+			TestUtils.waitForState(integrationsEndpoint, updatedIntegration, newDepState, TimeUnit.MINUTES, 10);
```

Comment:
Shouldn't there be an assert if the wait is successful?

Reply:
well, it should be indeed